trpcultivate_germplasm/src/Plugin/TripalImporter/GermplasmAccessionImporter.php

```diff
@@ -0,0 +1,885 @@
+            $this->error_tracker = TRUE;
+            return false;
+          }
+          else {
+            $synonym_id = $result;
+          }
+        }
+        // ------------------------------------------------------------------------
+        // Create a synonym-stock relationship via chado.stock_synonym
+        // ------------------------------------------------------------------------
+        // First check if this stock-synonym relationship already exists
+        $synonym_stock_query = $this->connection->select('1:stock_synonym', 'ss')
+          ->fields('ss', ['stock_synonym_id'])
+          ->condition('ss.synonym_id', $synonym_id, '=')
+          ->condition('ss.stock_id', $stock_id, '=');
+        $synonym_stock_record = $synonym_stock_query->execute()->fetchAll();
+
+        // Make sure there aren't 2 or more records
+        if (sizeof($synonym_stock_record) >= 2) {
+          $this->logger->error("Found more than one stock-synonym relationship for stock ID \"@stock\" and synonym \"@synonym\" in chado.stock_synonym.", ['@stock' => $stock_id, '@synonym' => $synonym]);
+          $this->error_tracker = TRUE;
+          return false;
+        }
+        // If 1 result was returned, just ignore it and move on
+        elseif (sizeof($synonym_stock_record) == 0) {
+          $values = [
+              'synonym_id' => $synonym_id,
+              'stock_id'=> $stock_id,
+              'pub_id'=> '1', // Set to the NULL publication and hopefully someone will update it later :)
+          ];
+          $result = $this->connection->insert('1:stock_synonym')
+            ->fields($values)
+            ->execute();
+
+          // If the primary key is not available, then the insert failed
+          if (!$result) {
+            $this->logger->error("Insertion of stock ID \"@stock\" and synonym \"@synonym\" into chado.stock_synonym failed.", ['@stock' => $stock_id, '@synonym' => $synonym]);
+            $this->error_tracker = TRUE;
+            return false;
+          }
+        }
+
+        // ------------------------------------------------------------------------
+        // Lastly, check if our synonym is in the stock table. If yes, THEN create
+        // a stock_relationship to connect this stock_id to the synonym.
+        // ------------------------------------------------------------------------
+        $stock_relationship_type_id = $this->getCVterm('stock_relationship_type_synonym');
+
+        $stock_query = $this->connection->select('1:stock', 'st')
+          ->fields('st', ['stock_id'])
+          ->condition('st.name', $synonym, '=')
+          ->condition('st.organism_id', $organism_id, '=');
+        $stock_record = $stock_query->execute()->fetchAll();
+
+        // Make sure there aren't 2 or more records
+        if (sizeof($stock_record) >= 2) {
+          $this->logger->error("Found more than one match for synonym name \"@synonym\" in chado.stock.", ['@synonym' => $synonym]);
```

Comment:
This is not actually an error since we didn't specify type. If this happens we just don't make a relationship. We do want to let them know but it's not an error.

Reply:
Thanks - addressed in commit [371d157](https://github.com/TripalCultivate/TripalCultivate-Germplasm/pull/6/commits/371d157617971458b627935d837d61c31d7aeae9)!